taichi/transforms/lower_matrix_ptr.cpp

```diff
@@ -0,0 +1,60 @@
+#include "taichi/ir/ir.h"
+#include "taichi/ir/statements.h"
+#include "taichi/ir/transforms.h"
+#include "taichi/ir/visitors.h"
+#include "taichi/system/profiler.h"
+
+TLANG_NAMESPACE_BEGIN
+
+class LowerMatrixPtr : public BasicStmtVisitor {
+ public:
+  using BasicStmtVisitor::visit;
+  DelayedIRModifier modifier;
+
+  void visit(PtrOffsetStmt *stmt) override {
+    if (stmt->origin->is<MatrixOfGlobalPtrStmt>()) {
+      TI_ASSERT(stmt->offset->is<ConstStmt>());
```

Comment:
What about the situation where a MatrixField is indexed with a LoopIndex? for example:
```
x = ti.Matrix.field(m=2, shape=(5))
@ti.kernel
def test():
  for i in x:
    for j in range(2):
       x[i][j] = 100
```
In this case, x[i] is MatrixOfGlobalPtrStmt, and x[i][j] is a PtrOffsetStmt with offset being `LoopIndexStmt`?

Reply:
This case requires dynamic index, which will be supported in a future PR :-)